### PR TITLE
chore: upgrade rust toolchain to latest nightly

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Make Intellij IDEs happy. Previous rust toolchain could cause the IDE falsely report "cannot use of moved value".

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by waynexia** [comment]
Be sure to update all CI yaml correspondingly

File: .cargo/config.toml [hunk 1]
```diff
@@ -13,4 +13,9 @@ rustflags = [
     "-Wclippy::print_stderr",
     "-Wclippy::implicit_clone",
     "-Aclippy::items_after_test_module",
+
+    # Strange clippy warnings when upgrading rust toolchain to nightly-2023-07-30.
+    # It will complain if the trait implementor has functions with `unimplemented!()` in it.
+    # Allow it for now; try disallow it when the toolchain is upgraded in the future.
+    "-Aclippy::diverging_sub_expression",
```
Comment (evenyag) [inline .cargo/config.toml:20]: Does an elder nightly version also have this error?

Comment (MichaelScofield) [reply .cargo/config.toml:20]: Don't know which version introduces this error, maybe some version in june.

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Comment by v0y4g3r** [comment]
Could you list the major changes between 2023-05-03 and 2023-07-30?

**Comment by MichaelScofield** [comment]
@v0y4g3r sorry, it seems that nightly rust does not have release notes. Can gpt do the work?

**Comment by v0y4g3r** [comment]
> @v0y4g3r sorry, it seems that nightly rust does not have release notes. Can gpt do the work?

Tried 🤣 but GPT4 failed at that work.

It seems the major change are:
- Empty struct initialization
- rustfmt supports let-else

File: .cargo/config.toml (hunk 1 above)
Comment (MichaelScofield) [reply .cargo/config.toml]: update the rust again and this false positive is gone, PTAL @evenyag

**Review by shuiyisong — APPROVED** [review]
Let's go

**Review by fengjiachun — APPROVED** [review]

File: .cargo/config.toml [hunk 2]
```diff
@@ -12,5 +12,9 @@ rustflags = [
     "-Wclippy::print_stdout",
     "-Wclippy::print_stderr",
     "-Wclippy::implicit_clone",
-    "-Aclippy::items_after_test_module",
+
+    # It seems clippy has made a false positive decision here when upgrading rust toolchain to
+    # nightly-2023-08-07, we do need it to be borrowed mutably.
+    # Allow it for now; try disallow it when the toolchain is upgraded in the future.
+    "-Aclippy::needless_pass_by_ref_mut",
```
Comment (evenyag) [inline .cargo/config.toml:19]: I guess this is related to https://github.com/rust-lang/rust/issues/114405

**Pull request merged**